control-userlayer/src/main/java/org/oskari/control/userlayer/UserLayerHandlerHelper.java

```diff
@@ -23,5 +29,48 @@ public static UserLayer getUserLayer(UserLayerDbService service, ActionParameter
         }
         return userLayer;
     }
-
+    public static JSONObject createErrorJSON (String error, String key, Set<String> set) {
+        JSONObject obj = createErrorJSON (error);
+        JSONHelper.put(obj, key, parseStringSet(set));
+        return obj;
+    }
+    public static JSONObject createErrorJSON (String error, String key, String value) {
+        JSONObject obj = JSONHelper.createJSONObject("error", error);
+        JSONHelper.putValue(obj, key, value);
+        return obj;
+    }
+    public static JSONObject createErrorJSON (String error) {
+        return JSONHelper.createJSONObject("error", error);
+    }
+    public static void addSetToErrorJSON (JSONObject obj, String key, Set<String> set) {
+        if (set.isEmpty()){
+            return;
+        }
+        if (obj == null) {
+            obj = new JSONObject();
+        }
+        JSONHelper.put(obj, key, parseStringSet(set));
+    }
+    public static void addMapToErrorJSON (JSONObject obj, String key, Map<String,String> map) {
+        if(map.isEmpty()){
+            return;
+        }
+        if (obj == null) {
+            obj = new JSONObject();
+        }
+        JSONHelper.putValue(obj, key, new JSONObject(map));
+    }
+    public static JSONObject addStringToErrorJSON (JSONObject obj, String key, String str) {
+        if (obj == null) {
+            obj = new JSONObject();
+        }
+        JSONHelper.putValue(obj, key, str);
+        return obj;
+    }
+    public static JSONArray parseStringSet (Set<String> set){
```

Comment:
This isn't required, can use `new JSONArray(Collection collection)` constructor for the same effect.